spaceship/jobs/base_job.py

```diff
@@ -0,0 +1,53 @@
+import logging
+
+from rq import Queue
+
+from .. import app
+from ..redis import store
+
+class BaseJob:
+  """Convenience class to make delayed work with RQ easier
+
+  Derive from this class and implement the classmethod `perform()`. You can
+  execute jobs directly by calling `.perform()`, or call `.enqueue()` to send
+  the work to a worker process.
+  """
+
+  # Default RQ settings for this job -- override in child classes
+  QUEUE = 'default'    # what queue will the job go on?
+  TIMEOUT_SECS = None  # how long to allow the job to run before failing it?
+  TTL_SECS = -1        # how long to keep the job in the queue before giving up?
+  RESULT_TTL_SECS = 500  # how long to keep the results in Redis?
+
+  @classmethod
+  def enqueue(cls, *args, **kwargs):
+    q = Queue(
+      cls.QUEUE,
+      connection=store,
+
+      # in dev, we won't use queues but invoke the job here-and-now
+      is_async=app.config['IN_PRODUCTION'],
+    )
+
+    job = q.enqueue_call(
+      func=cls.perform,
```

Comment:
So to check my understanding, cls.perform will be called in the context of an rq worker, right? And so this is a serialization of the call; globals etc from the environment are not available from job classes?

Reply:
Hopefully our peewee models will serialize correctly, I can see us trying to pass those around.

Reply:
i think it's just serializing the call and the parameters, not the code that will be invoked.

Reply:
Right yeah I'm wondering what happens when the parameter is a peewee object. Do we pass... the full current state of the db object at call time? Enough info to refetch it? Does an orm cache in the worker context hold on to stale models and cause consistency issues? What if a migration happens with pending jobs that reference old models... There are just all kinds of exciting ways this can break. :)

Reply:
i was wrong, it does try to serialize class instances. we should probably avoid doing that...

Reply:
yeah i'm not a fan of the pickling of instances, i assumed it serialized simple data along with references by name. i'll see if i can constrain the world a bit here